modules/hpe_nimble_volume.py

```diff
@@ -0,0 +1,897 @@
+        - Set volume's provisioning level to thin.
+    encryption_cipher:
+        required: False
+        choices:
+        - none
+        - aes_256_xts
+        type: str
+        default: none
+        description:
+        - The encryption cipher of the volume.
+    app_uuid:
+        required: False
+        type: str
+        description:
+        - Application identifier of volume.
+    folder:
+        required: False
+        type: str
+        description:
+        - Name of the folder holding this volume.
+    metadata:
+        required: False
+        type: dict
+        description:
+        - Key-value pairs that augment an volume's attributes.
+    dedupe_enabled:
+        required: False
+        type: bool
+        default: False
+        description:
+        - Indicate whether dedupe is enabled.
+    limit_iops:
+        required: False
+        type: int
+        default: -1
+        description:
+        - IOPS limit for this volume.
+    limit_mbps:
+        required: False
+        type: int
+        default: -1
+        description:
+        - Throughput limit for this volume in MB/s.
+    parent:
+        required: False
+        type: str
+        description:
+        - Name of parent volume.
+    snapshot:
+        required: False
+        type: str
+        description:
+        - Base snapshot name. This attribute is required together with name and clone when cloning a volume with the create operation.
+    volcoll:
+        required: False
+        type: str
+        description:
+        - Name of volume collection of which this volume is a member.
+    force:
+        required: False
+        type: bool
+        default: False
+        description:
+        - Forcibly offline, reduce size or change read-only status a volume.
+    caching_enabled:
+        required: False
+        type: bool
+        default: False
+        description:
+        - Indicate caching the volume is enabled.
+    force_vvol:
+        required: False
+        type: bool
+        default: False
+        description:
+        - Forcibly move a Virtual volume.
+    move:
+        required: False
+        type: bool
+        default: False
+        description:
+        - Move a volume to different Pool.
+
+extends_documentation_fragment: hpe_nimble
+short_description: "Manages a HPE Nimble Storage volume"
+version_added: "2.9"
+'''
+
+EXAMPLES = r'''
+
+    # If state is "create", then create a volume if not present. Fails if already present.
+    # If state is "present", then create a volume if not present. Succeeds if it already exists.
+    - name: Create volume if not present
+      hpe_nimble_volume:
+        hostname: "{{ hostname }}"
+        username: "{{ username }}"
+        password: "{{ password }}"
+        state: "{{ state | default('present') }}" # fail if exist
+        size: "{{ size }}"
+        limit_iops: "{{ limit_iops }}"
+        limit_mbps: 5000
+        force: false
+        metadata: "{{ metadata }}"
```

Comment:
Don't we need to elaborate a bit further what metadata contains? I can see we put it in the variable table but an example would be helpful I think.

Reply:
is this ok ??
`
metadata: "{{ metadata }}" # metadata = {'mykey1': 'myval1', 'mykey2': 'myval2'}`